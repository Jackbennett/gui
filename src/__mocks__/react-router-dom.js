import React from 'react';
import { compose, setDisplayName } from 'recompose';
import * as rrd from 'react-router-dom';
// Just render plain div with its children
// eslint-disable-next-line no-import-assign
rrd.BrowserRouter = compose(setDisplayName('BrowserRouter'))(({ children }) => <div>{children}</div>);

module.exports = rrd;
