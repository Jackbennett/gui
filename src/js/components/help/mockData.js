export const helpProps = {
  docsVersion: '',
  findLocalIpAddress: jest.fn,
  isHosted: false,
  isEnterprise: false,
  menderDebPackageVersion: 'master',
  menderVersion: 'master',
  menderArtifactVersion: 'master',
  org: {
    tenant_token: ''
  }
};
