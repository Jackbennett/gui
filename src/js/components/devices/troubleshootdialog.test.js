import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';
import TroubleshootDialog from './troubleshootdialog';
import { defaultState, undefineds } from '../../../../tests/mockData';

const mockStore = configureStore([thunk]);

describe('TroubleshootDialog Component', () => {
  let store;
  let socketSpyFactory;
  let socketSpy;
  const oldMatchMedia = window.matchMedia;

  beforeEach(() => {
    store = mockStore({ ...defaultState });
    socketSpyFactory = jest.spyOn(window, 'WebSocket');
    socketSpyFactory.mockImplementation(() => {
      socketSpy = {
        close: () => {},
        send: () => {}
      };
      return socketSpy;
    });
    Object.defineProperty(window, 'matchMedia', {
      writable: true,
      value: jest.fn().mockImplementation(query => ({
        matches: false,
        media: query,
        onchange: null,
        addListener: jest.fn(), // Deprecated
        removeListener: jest.fn(), // Deprecated
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn()
      }))
    });
  });

  afterEach(() => {
    socketSpyFactory.mockReset();
    window.matchMedia = oldMatchMedia;
  });

  it('renders correctly', async () => {
    const { baseElement } = render(
      <MemoryRouter>
        <Provider store={store}>
          <TroubleshootDialog onCancel={jest.fn} onSocketClose={jest.fn} open={true} />
        </Provider>
      </MemoryRouter>
    );
    const view = baseElement.getElementsByClassName('MuiDialog-root')[0];
    expect(view).toMatchSnapshot();
    expect(view).toEqual(expect.not.stringMatching(undefineds));
  });
});
