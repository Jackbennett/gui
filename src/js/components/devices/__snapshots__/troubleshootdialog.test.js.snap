// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TroubleshootDialog Component renders correctly 1`] = `
<div
  class="MuiDialog-root"
  role="presentation"
  style="position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px;"
>
  <div
    aria-hidden="true"
    class="MuiBackdrop-root"
    style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
  />
  <div
    data-test="sentinelStart"
    tabindex="0"
  />
  <div
    class="MuiDialog-container MuiDialog-scrollPaper"
    role="none presentation"
    style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    tabindex="-1"
  >
    <div
      class="MuiPaper-root MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthLg MuiDialog-paperFullWidth MuiPaper-elevation24 MuiPaper-rounded"
      role="dialog"
    >
      <div
        class="MuiDialogTitle-root"
      >
        <h2
          class="MuiTypography-root MuiTypography-h6"
        >
          Troubleshoot
        </h2>
      </div>
      <div
        class="MuiDialogContent-root dialog-content flexbox column"
        style="padding: 0px; margin: 0px 24px; height: 75vh;"
      >
        <div
          class="MuiTabs-root"
        >
          <div
            class="MuiTabs-scroller MuiTabs-fixed"
            style="overflow: hidden;"
          >
            <div
              class="MuiTabs-flexContainer"
              role="tablist"
            >
              <button
                aria-selected="true"
                class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected"
                role="tab"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiTab-wrapper"
                >
                  <div
                    class="flexbox"
                    style="align-items: center;"
                  >
                    Remote terminal
                     
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                      focusable="false"
                      style="color: rgb(143, 13, 13); margin-left: 8px;"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M20,19V7H4V19H20M20,3A2,2 0 0,1 22,5V19A2,2 0 0,1 20,21H4A2,2 0 0,1 2,19V5C2,3.89 2.9,3 4,3H20M13,17V15H18V17H13M9.58,13L5.57,9H8.4L11.7,12.3C12.09,12.69 12.09,13.33 11.7,13.72L8.42,17H5.59L9.58,13Z"
                      />
                    </svg>
                  </div>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
            <span
              class="PrivateTabIndicator-root-1 PrivateTabIndicator-colorSecondary-3 hidden"
              style="left: 0px; width: 0px;"
            />
          </div>
        </div>
        <div
          class=""
          style="display: grid; grid-template-rows: max-content 0; flex-grow: 1; overflow: hidden;"
        >
          <div
            class="margin-top-small margin-bottom-small"
          >
            <div>
              <b>
                Session status:
              </b>
               
              disconnected
            </div>
            <div>
              <b>
                Connection start:
              </b>
               
              -
            </div>
            <div>
              <b>
                Duration:
              </b>
               
              00:00:00
            </div>
          </div>
          <div
            style="position: relative; max-height: 0; overflow: hidden;"
            tabindex="0"
          >
            <div
              style="position: absolute; width: 100%; height: 100%; max-height: 0; overflow: hidden;"
            >
              <div
                class="terminal xterm"
                dir="ltr"
                role="document"
                tabindex="0"
              >
                <div
                  class="xterm-viewport"
                  style="background-color: rgb(0, 0, 0);"
                >
                  <div
                    class="xterm-scroll-area"
                  />
                </div>
                <div
                  class="xterm-screen"
                >
                  <div
                    class="xterm-helpers"
                  >
                    <textarea
                      aria-label="Terminal input"
                      aria-multiline="false"
                      autocapitalize="off"
                      autocorrect="off"
                      class="xterm-helper-textarea"
                      spellcheck="false"
                      tabindex="0"
                    />
                    <span
                      aria-hidden="true"
                      class="xterm-char-measure-element"
                      style="font-family: courier-new, courier, monospace; font-size: 15px;"
                    >
                      W
                    </span>
                    <div
                      class="composition-view"
                    />
                  </div>
                  <canvas
                    class="xterm-text-layer"
                    style="z-index: 0;"
                  />
                  <canvas
                    class="xterm-selection-layer"
                    style="z-index: 1;"
                  />
                  <canvas
                    class="xterm-link-layer"
                    style="z-index: 2;"
                  />
                  <canvas
                    class="xterm-cursor-layer"
                    style="z-index: 3;"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="flexbox centered"
            style="background: rgba(0, 0, 0, 0.8);"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                Connect Terminal
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        class="MuiDialogActions-root flexbox space-between MuiDialogActions-spacing"
      >
        <div>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Connect
               Terminal
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <a
            aria-disabled="false"
            class="MuiButtonBase-root MuiButton-root MuiButton-text"
            href="/auditlog?object_id=undefined&start_date=2020-01-01T00:00:00.000Z"
            role="button"
            tabindex="0"
          >
            <span
              class="MuiButton-label"
            >
              View 
              session logs
               for this device
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </a>
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Close
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
  <div
    data-test="sentinelEnd"
    tabindex="0"
  />
</div>
`;
